### ISSUE TITLE
mod_wsgi error for length of path for daemon process socket

### ISSUE THREAD

**Issue by pmc2010** [GrahamDumpleton/mod_wsgi#288]
I'm getting this error upon startup of mod_wsgi-express:

[Sat Mar 31 00:57:03.787544 2018] [wsgi:alert] [pid 55899] mod_wsgi (pid=55899): Length of path for daemon process socket exceeds maxmimum allowed value and will be truncated, resulting in likely failure to bind the socket, or other later related failure.
[Sat Mar 31 00:57:03.788458 2018] [mpm_prefork:notice] [pid 55899] AH00163: Apache/2.4.28 (Unix) mod_wsgi/4.6.3 Python/2.7 configured -- resuming normal operations
[Sat Mar 31 00:57:03.788483 2018] [core:notice] [pid 55899] AH00094: Command line: 'httpd (mod_wsgi-express)        -f /var/folders/x_/h3h5r8qn5tq3rvyt21259lt038yvfj/T/mod_wsgi-localhost:8080:110063057/httpd.conf -D MOD_WSGI_LOAD_PYTHON_DYLIB -D FOREGROUND'

Any idea what this means or where I should look to find out more?  The site will work momentarily, and then I'll get a page saying "My web site runs on Malt Whiskey."

**Comment by GrahamDumpleton** [comment]
Use the ``--server-root`` option to specify a different directory into which generated files are generated. The uid you are using is pushing the full pathname of the directory created over the limit allowed for a UNIX socket path.

**Comment by pmc2010** [comment]
Hmm... I navigated to the root folder and tried this command:

mod_wsgi-express start-server --port 8080 --document-root .

and this command:
mod_wsgi-express start-server --port 8080 --document-root /

and still got the error message

**Comment by pmc2010** [comment]
hold up, wrong command.

**Comment by GrahamDumpleton** [comment]
The ``--server-root`` option not ``--document-root``. You shouldn't use ``/`` for either.

So:

```
mod_wsgi-express --server-root /tmp/wsgi
```

You shouldn't use under ``/tmp`` though as cron on some system will remove files from there which haven't been modified for a while and break things.

**Comment by pmc2010** [comment]
Got it, that worked. Thank you for the quick response!

**Comment by GrahamDumpleton** [comment]
I need to look at changing to use ``/var/tmp`` as parent for server root. I did originally use ``/tmp``, but some cron systems remove files which are old in ``/tmp``. This is why I moved to using ``$TMPDIR`` if set, but on MacOS X at least, that results in a long directory path. Up till now that hasn't caused a problem, but looks like is close to limit and the long uid can push it over. I should be okay using just ``/var/tmp`` as I qualify the sub directory created by user ID and port anyway.
